x-pack/platform/plugins/shared/alerting/common/routes/alert_delete/preview/schemas/v1.ts

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { schema } from '@kbn/config-schema';
+import { alertDeleteCategoryIds } from '../../../../constants';
+
+export const alertDeletePreviewQuerySchema = schema.object({
+  is_active_alert_delete_enabled: schema.boolean({
+    meta: {
+      description: 'Enable deletion of active alerts when set to true',
+    },
+  }),
+  active_alert_delete_threshold: schema.number({
+    min: 1,
+    max: 1000,
+    meta: {
+      description:
+        'Threshold (in days) for deleting active alerts older than this value, applies only when deletion is enabled',
+    },
+  }),
+  is_inactive_alert_delete_enabled: schema.boolean({
+    meta: {
+      description:
+        'Enable deletion of inactive alerts (recovered/closed/untracked) when set to true',
+    },
+  }),
+  inactive_alert_delete_threshold: schema.number({
+    min: 1,
+    max: 1000,
+    meta: {
+      description:
+        'Threshold (in days) for deleting inactive alerts (recovered/closed/untracked) older than this value, applies only when deletion is enabled',
+    },
+  }),
+  category_ids: schema.maybe(
```

Comment:
should we make this required?

Reply:
I plan to do that in another PR, there are some tests making sure everything is deleted if this is not set